shanghai/client.py

```diff
@@ -21,6 +21,13 @@ def __init__(self, host, port, queue: asyncio.Queue, ssl=False, loop=None):
     def sendline(self, line):
         self.writer.write(line.encode('utf-8') + b'\r\n')
 
+    def sendcmd(self, command, params=None):
+        params = [] if params is None else params
+        args = [command] + params
+        if ' ' in args[-1]:
```

Comment:
No, per RFC, spaces are used to separate parameters from each other. So technically it's an error, if command or any of the other arguments have spaces.

Well I see... if it happens that command has spaces (for some reason), and params is empty, it would yield an invalid irc line... something like `":THE COMMAND"` if command is `"THE COMMAND"`, which would be interpreted as prefix `"THE"` and command `"COMMAND"` by the server.

I was just too lazy to do it properly :-) There should probably be some sanity checks, so that plugin writers don't call those methods with invalid values.